plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeTableChangesWithAccessTracking.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.Session;
+import io.trino.filesystem.hdfs.HdfsFileSystemFactory;
+import io.trino.plugin.deltalake.metastore.TestingDeltaLakeMetastoreModule;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.Optional;
+
+import static com.google.inject.util.Modules.EMPTY_MODULE;
+import static io.trino.plugin.hive.HiveTestUtils.HDFS_ENVIRONMENT;
+import static io.trino.plugin.hive.metastore.file.FileHiveMetastore.createTestingFileHiveMetastore;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Test(singleThreaded = true)
+public class TestDeltaLakeTableChangesWithAccessTracking
+        extends AbstractTestQueryFramework
+{
+    protected static final String SCHEMA = "test_table_changes_tracking_" + randomNameSuffix();
+
+    private static final Session TEST_SESSION = testSessionBuilder()
+            .setCatalog("delta_lake")
+            .setSchema(SCHEMA)
+            .build();
+
+    private File dataDirectory;
+    private HiveMetastore metastore;
```

Comment:
My IDE warns me that "Field can be converted to a local variable "

BTW why we provide metastoreModule? you can use normal configuration for that, like we do https://github.com/trinodb/trino/blob/1a8cacad2dfa2553467041e897a36fd1a26cffb7/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/DeltaLakeQueryRunner.java#L109-L111

Reply:
I converted fields to local variables.
Regarding the second thing the reason is that here we want to call `io.trino.plugin.deltalake.TestingDeltaLakePlugin` constructor that accepts FileSystemFactory, that constructor also requires MetastoreModule etc.